src/main/java/uk/gov/nationalarchives/pdi/step/xml/CanonicalStepDialog.java

```diff
@@ -0,0 +1,283 @@
+     * must remain unaltered.
+     *
+     * The open() method must return the name of the step after the user has confirmed the dialog,
+     * or null if the user cancelled the dialog.
+     */
+    public String open() {
+        // store some convenient SWT variables
+        Shell parent = getParent();
+        Display display = parent.getDisplay();
+
+        // SWT code for preparing the dialog
+        shell = new Shell( parent, SWT.DIALOG_TRIM | SWT.RESIZE | SWT.MIN | SWT.MAX );
+        props.setLook( shell );
+        setShellImage( shell, meta );
+
+        // Save the value of the changed flag on the meta object. If the user cancels
+        // the dialog, it will be restored to this saved value.
+        // The "changed" variable is inherited from BaseStepDialog
+        changed = meta.hasChanged();
+
+        // The ModifyListener used on all controls. It will update the meta object to
+        // indicate that changes are being made.
+        ModifyListener lsMod = new ModifyListener() {
+            public void modifyText( ModifyEvent e ) {
+                meta.setChanged();
+            }
+        };
+
+        // ------------------------------------------------------- //
+        // SWT code for building the actual settings dialog        //
+        // ------------------------------------------------------- //
+        FormLayout formLayout = new FormLayout();
+        formLayout.marginWidth = Const.FORM_MARGIN;
+        formLayout.marginHeight = Const.FORM_MARGIN;
+        shell.setLayout( formLayout );
+        shell.setText( BaseMessages.getString( PKG, "CanonicalStep.Shell.Title" ) );
+        int middle = props.getMiddlePct();
+        int margin = Const.MARGIN;
+
+        // Stepname label
+        wlStepname = new Label( shell, SWT.RIGHT );
+        wlStepname.setText( BaseMessages.getString( PKG, "System.Label.StepName" ) );
+        props.setLook( wlStepname );
+        fdlStepname = new FormData();
+        fdlStepname.left = new FormAttachment( 0, 0 );
+        fdlStepname.right = new FormAttachment( middle, -margin );
+        fdlStepname.top = new FormAttachment( 0, margin );
+        wlStepname.setLayoutData( fdlStepname );
+
+        // Stepname input box
+        wStepname = new Text( shell, SWT.SINGLE | SWT.LEFT | SWT.BORDER );
+        wStepname.setText( stepname );
+        props.setLook( wStepname );
+        wStepname.addModifyListener( lsMod );
+        fdStepname = new FormData();
+        fdStepname.left = new FormAttachment( middle, 0 );
+        fdStepname.top = new FormAttachment( 0, margin );
+        fdStepname.right = new FormAttachment( 100, 0 );
+        wStepname.setLayoutData( fdStepname );
+
+        // FieldName to evaluate
+        Label wlField = new Label(shell, SWT.RIGHT);
+        wlField.setText( BaseMessages.getString( PKG, "CanonicalStep.Input.Field.Label" ) );
+        props.setLook(wlField);
+        FormData fdlField = new FormData();
```

Comment:
Is it easier/clearer to use the Builder API here like in the other plugins? e.g.

```
FormData fdlField = new FormDataBuilder()
                .left(0, 0)
                .top(wStepname, 2 * margin)
                .right(middle, -margin)
                .result();
```